### PR TITLE
feat: Delete album

### PR DESCRIPTION
**Pull request by rajsam003**
This PR resolves [MOPLAT-315]

https://user-images.githubusercontent.com/17421923/177986388-bbddff66-1d8f-4d73-a1dc-d0ccbe8d149e.mov

### PR Checklist

- [x] I tested my changes on **iOS** and **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-315]: https://artsyproduct.atlassian.net/browse/MOPLAT-315?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx [hunk 1]
```diff
@@ -1,27 +1,61 @@
...
 
 export const AlbumArtworks = () => {
   const { albumId } = useRoute<AlbumArtworksRoute>().params
+  const navigation = useNavigation<NavigationProp<HomeTabsScreens>>()
+  const safeAreaInsets = useSafeAreaInsets()
   const albums = GlobalStore.useAppState((state) => state.albums.albums)
   const album = albums.find((album) => album.id === albumId)
 
+  const deleteAlbumHandler = () =>
+    Alert.alert("Do you want to delete this album?", "You cannot undo this action", [
```
Comment (admbtlr) [inline src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx:21]: Can you change this to "Are you sure you want to delete this album?"

File: src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx [hunk 2]
```diff
@@ -1,27 +1,61 @@
...
+      {
+        text: "Cancel",
+        style: "cancel",
+      },
+      {
+        text: "Delete",
+        style: "destructive",
+        onPress: async () => {
+          await GlobalStore.actions.albums.removeAlbum(album?.id || "")
```
Comment (pvinis) [inline src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx:30]: can we remove all these `|| ""`? they will only make it harder in the future.

File: src/app/sharedUI/Header.tsx [hunk 3]
```diff
@@ -24,7 +24,7 @@ export const Header = ({ label, withoutBackButton, rightElements }: HeaderProps)
       ) : null}
       <Flex flexDirection="row" mt={2} alignItems="center" justifyContent="space-between">
         <Text variant="lg">{label}</Text>
-        <Flex>{rightElements}</Flex>
```
[the remark this answers is not in the document]
Comment (rajsam003) [reply src/app/sharedUI/Header.tsx:27]: No its not

File: src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx [hunk 4]
```diff
@@ -1,27 +1,61 @@
...
+import { Alert } from "react-native"
 
 type AlbumArtworksRoute = RouteProp<HomeTabsScreens, "AlbumArtworks">
 
 export const AlbumArtworks = () => {
   const { albumId } = useRoute<AlbumArtworksRoute>().params
+  const navigation = useNavigation<NavigationProp<HomeTabsScreens>>()
+  const safeAreaInsets = useSafeAreaInsets()
   const albums = GlobalStore.useAppState((state) => state.albums.albums)
   const album = albums.find((album) => album.id === albumId)
```
Comment (pvinis) [inline src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx:18]: by adding a check here maybe, if album is null then return null or something, and that means that anywhere after this check, we don't need to do `album?.something`.

do you wanna try this?

File: src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx (hunk 2 above)
Comment (rajsam003) [reply src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx:30]: I am getting this type issue then.
image

File: src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx (hunk 4 above)
Comment (rajsam003) [reply src/app/screens/HomeTabs/Albums/AlbumArtworks.tsx:18]: Good idea

**Review by pvinis — APPROVED** [review]

**Pull request merged**